web/api/v1/api.go

```diff
@@ -491,7 +491,8 @@ func (api *API) series(r *http.Request) apiFuncResult {
 	var sets []storage.SeriesSet
 	var warnings storage.Warnings
 	for _, mset := range matcherSets {
-		s, wrn, err := q.Select(nil, mset...) //TODO
+		// SelectParams.Metadata=true should instruct remote_read endpoint to send labels only (without samples)
+		s, wrn, err := q.Select(&storage.SelectParams{Metadata: true}, mset...)
```

Comment:
My concern is that this line specifically will cause anyone using the old check (if nil) to start fetching all data.

Reply:
We can mention it as a CHANGE in the changelog.

Reply:
IMO simply changing from having to check whether its a nil to checking a single field for a bool doesn't seem like enough improvement (if any) to justify the breaking change. If this were going to start adding *more* information I could maybe be onboard, but at this point its moving a boolean check from the current "is nil" to this "metadata" field -- which IMO seems like an unnecessary change (which isn't providing any value-- at least not that I'm seeing in this PR).

Reply:
I tend to agree with @jacksontj . The main benefit of this commit is that it brings more clarity to what the remote_read endpoint would expect to do. There is no functional change in the code.

Originally I intended to introduce a new method to the `Querier` interface for querying the metrics (labels). I actually have that change in my other branch. But then I realized that the new method will have almost identical signature with the `Select` method: it will receive a list of `labels.Matcher`s and return a `SeriesSet`. So I dropped that idea.

@brian-brazil @jacksontj leaving it up to you guys.

Reply:
Sounds like maybe the solution for now is to better document the `ReadHints` + `SelectParams` nil meaning?

Reply:
That makes sense.